crates/bevy_ecs/src/system/system.rs

```diff
@@ -94,6 +94,32 @@ pub trait System: Send + Sync + 'static {
     /// of this system into the world's command buffer.
     fn queue_deferred(&mut self, world: DeferredWorld);
 
+    /// Validates that all systems parameters can be acquired.
+    /// If not, the system should not be ran.
```

Comment:
This should however mention that the implementation of `System` should not rely on this fact being respected for safety. That is, safe code can still call `System::run` even if `System::validate_param` return `false` and that should panic rather than causing UB.